apps/system/test/unit/task_manager_test.js

```diff
@@ -1,1377 +1,570 @@
-      target.dispatchEvent(new CustomEvent('transitionend'));
-      this.sinon.clock.tick(100);
-    });
-  });
-
-  suite('closeApp', function() {
-    setup(function() {
-      MockStackManager.mStack = [
-        appConfigs.sms,
-        appConfigs.game
-      ].map((config) => {
-        var app = new AppWindow(config);
-        apps[app.name] = app;
-        return app;
-      });
-      MockStackManager.mCurrent = 0;
-      showTaskManager(this.sinon.clock);
-      this.sinon.clock.tick();
-    });
+      MockService.mockQueryWith('getTopMostWindow', apps.search);
+      MockStackManager.mCurrent = MockStackManager.mStack.length - 1;
 
-    test('removes the card for that app', function() {
-      this.sinon.stub(taskManager, 'removeCard');
-      sendAppTerminated(apps.sms);
-      assert.isTrue(taskManager.removeCard.calledOnce);
+      tm = new TaskManager();
+      tm.start().then(() => tm.show()).then(() => { done(); }, done);
+      clock.tick(1000);
     });
 
-    test('destroys the card', function() {
-      var card = taskManager.getCardAtIndex(0);
-      var destroySpy = this.sinon.spy(card, 'destroy');
-      assert.isTrue(card && card.element &&
-                    card.element.parentNode == taskManager.cardsList);
-      var instanceID = card.app.instanceID;
-      sendAppTerminated(card.app);
-      assert.isTrue(destroySpy.calledOnce);
-      assert.equal(cardsList.children.length, 1);
-      assert.isFalse(instanceID in taskManager.cardsByAppID);
+    teardown(function(done) {
+      tm.hide().then(() => tm.stop()).then(done, done);
+      clock.tick(1000);
     });
-    suite('after destroying all the cards', function() {
-      setup(function() {
-        sendAppTerminated(apps.sms);
-        sendAppTerminated(apps.game);
-      });
 
-      test('should go back home', function(done) {
-        var stub = this.sinon.stub(home, 'open');
+    test('Proper state', (done) => {
+      assert.isTrue(tm.isShown());
+      assert.ok(document.querySelector('#cards-view.active:not(.empty)'));
 
-        waitForEvent(window, 'cardviewclosed').then(function() {
-          assert.isTrue(stub.calledOnce, 'home was open');
-        }, failOnReject)
-        .then(function() { done(); }, done);
+      assert.equal(
+        MockStackManager.getCurrent(),
+        tm.currentCard.app);
 
-        sendAppTerminated(apps.game);
-        fakeFinish(this.sinon.clock, home);
-      });
+      done();
     });
-  });
 
-  suite('app is killed', function() {
-    setup(function() {
-      MockStackManager.mStack = [
-        appConfigs.sms,
-        appConfigs.game
-      ].map((config) => {
-        var app = new AppWindow(config);
-        apps[app.name] = app;
-        return app;
+    test('Wheel handling for accessibility', (done) => {
+      var card = tm.currentCard;
+      var killAppStub = sinon.stub(card.app, 'kill');
+      sinon.stub(card.app, 'killable', () => true);
+      tm.handleEvent({
```

Comment:
if this event is hard to synthesized I can live with a direct `handleEvent` call here :)